examples/progress_bar_minimal.py

```diff
@@ -0,0 +1,84 @@
+"""This file provides minimal working examples of progress bars in
+the napari viewer.
+"""
+
+import napari
+from time import sleep
+from napari.utils.progress import progress
+from qtpy.QtWidgets import QPushButton, QVBoxLayout, QWidget
+import numpy as np
+
+def iterable():
+    """progress can be used as a wrapper around an iterable object
+    """
+    my_stacked_volume = np.random.random((5, 4, 500, 500))    
+    # progress provides a context manager we can use for automatic
+    # teardown of our widget once iteration is complete
+    with progress(my_stacked_volume) as pbr:
+        for i, im_slice in enumerate(pbr):
+            # using a context manager also allows us to manipulate
+            # the progress object e.g. by setting a description
+            pbr.set_description(f"Slice {i}")
+            sleep(0.4)            
+
+            # finally, only when using progress with a context manager
+            # can we make use of nested progress bars and ensure they
+            # are grouped together nicely in the viewer
+            for _ in progress(im_slice, desc="Channels"):
+                sleep(0.1)
+                
+def indeterminate():
+    """By passing a total of 0, we can have an indeterminate progress bar
+    """
+    pbr = progress(total=0)
+
+    for i in range(int(1e4)):
+        pbr.set_description(f"Processing {i}")
+
+    # if not using progress in a context manager or through a for loop
+    # we have to manually close it
+    pbr.close()
```

Comment:
let's always use the context manager unless it literally can't accomplish what we need.  
also suggest changing this example to something that is actually indeterminate (not just long)

```suggestion
    from random import choice

    with progress(total=0) as pbar:
        x = 0
        while x != 42:
            pbar.set_description(f"Processing {x}")
            x = choice(range(100))
            sleep(0.05)
```

Reply:
also, i was a little surprised that passing `progress(total=0)` gave this display:

<img width="472" alt="total0" src="https://user-images.githubusercontent.com/1609449/116487008-ee18ea80-a85c-11eb-9137-eba65e045ed2.png">

while passing `progress()` (no arguments) gave this:

<img width="472" alt="nototal" src="https://user-images.githubusercontent.com/1609449/116487018-f6712580-a85c-11eb-8e2d-879d988542e4.png">

is there a use case where passing no arguments should be different than total=0?

Reply:
Nope, this was a bug. Thanks for catching!